src/theory/arith/theory_arith_private.cpp

```diff
@@ -4821,8 +4815,11 @@ bool TheoryArithPrivate::propagateCandidateRow(RowIndex ridx){
   Debug("arith::prop")
     << "propagateCandidateRow " << instance << " attempt " << rowLength << " " <<  hasCount << endl;
 
-  if(rowLength >= options::arithPropagateMaxLength()){
-    if(fRand(0.0,1.0) >= double(options::arithPropagateMaxLength())/rowLength){
+  if (rowLength >= options::arithPropagateMaxLength())
+  {
+    if (RNG::getRNG().pickDouble(0.0, 1.0)
```

Comment:
This also seems like a case for `pickWithProb`.

Reply:
Again, @timothy-king can you comment on this?

Reply:
Ditto